crates/block-producer/src/mempool/mod.rs

```diff
@@ -73,32 +77,161 @@ impl BlockNumber {
     }
 }
 
+// MEMPOOL BUDGET
+// ================================================================================================
+
+/// Limits placed on a batch's contents.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct BatchBudget {
+    /// Maximum number of transactions allowed in a batch.
+    pub transactions: usize,
+    /// Maximum number of input notes allowed.
+    pub input_notes: usize,
+    /// Maximum number of output notes allowed.
+    pub output_notes: usize,
+    /// Maximum number of updated accounts.
+    pub accounts: usize,
```

Comment:
I saw that on `TransactionGraph` you do access `transactions`, maybe an accessor could be added

Reply:
To my mind these are a form of configuration where the default and production values just happen to be these other constants.

In a more fleshed out system I would allow node operators to override these values via the cli/config file. This is useful if we need to change these, either in production or for integration test purposes, without requiring a full re-release. The same should also be done with block / batch intervals for example.

> Do these need to be pub? 

More to your actual point though, these could be made private which would just make construction during configuration more awkward.

Reply:
Ah actually I guess your concern is that they're `pub` and the update requires `mut` access? Which could allow misuse, and means it isn't just a pure config anymore.. Fair enough, I'll make them private.